Celeste.Mod.mm/Patches/TextMenu.cs

```diff
@@ -15,11 +15,84 @@ public class patch_TextMenu : TextMenu {
         // We're effectively in TextMenu, but still need to "expose" private fields to our mod.
         private List<Item> items;
 
+        private bool batchMode;
+        private float width;
+        private float leftColumnWidth;
+        private float rightColumnWidth;
+        private float height;
+        private bool recalculatingSizeInBatchMode;
+
         /// <summary>
         /// The items contained in this menu.
         /// </summary>
         public List<Item> Items => items;
 
+        /// <summary>
+        /// When a menu is in batch mode, adding / removing items will not recalculate its size to improve performance.
+        /// Size is recalculated immediately after batch mode is disabled.
+        /// </summary>
+        public bool BatchMode {
+            get => batchMode;
+            set {
+                batchMode = value;
+                if (!batchMode) {
+                    RecalculateSize();
+                }
+            }
+        }
+
+        /// <inheritdoc cref="TextMenu.Width"/>
+        public new float Width {
+            [MonoModReplace]
+            get {
+                if (batchMode && !recalculatingSizeInBatchMode) {
+                    RecalculateSize();
+                }
+                return width;
+            }
+            [MonoModReplace]
+            private set => width = value;
+        }
```

Comment:
Does replacing this also remove the compiler-generated backing field?

Reply:
It doesn't - it'll simply go unused, which is acceptable, given how it's inaccessible anyway.